pkg/nodediscovery/nodediscovery.go

```
@@ -210,10 +210,15 @@ func (n *NodeDiscovery) createCiliumNodeResource(conf Configuration) {
 
 	ciliumClient := k8s.CiliumClient()
 
-	nodeResource := &ciliumv2.CiliumNode{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: node.GetName(),
-		},
+	performUpdate := true
+	nodeResource, err := ciliumClient.CiliumV2().CiliumNodes().Get(node.GetName(), metav1.GetOptions{})
+	if err != nil {
```

Will the error here only result if the resource did not exist before? That is now I am interpreting the code - would be helpful to have a comment confirming this here.

---

There can be other errors but I'm not sure what would be a better error resolution. A blind `Delete()`? Maybe we should make it fail hard so the pod itself restarts.

---

Changed to `Fatal()` logs to force retrying via pod restart

---

I'm ambivalent. I just wasn't sure about the semantics behind the returned error.